client/hack/README.md

```diff
@@ -54,6 +54,8 @@ Once you run the script, you will get an output as follows:
 
 ## update-crd.sh
 
+NOTE: We need to keep both v1beta1 and v1 snapshot APIs but set served and storage version of v1beta1 to false. Please copy back the v1beta1 manifest back to the files as this script will remove it.
+
```

Comment:
Shouldn't that be fixed in the script?

Reply:
@msau42 brought this to my attention.  See discussion here: https://github.com/kubernetes/kubernetes/pull/101694#issuecomment-831235149

Reply:
We have already deleted v1beta1 from API definition.  So the auto generated code will not include v1beta1 any more.  We just keep the v1beta1 manifest identical as v5.0.1 and will not change it.

Reply:
Every time you run the script, it also generates some changes such as new indentations even if there is no concrete change. It is hard to review them.  Just keeping v1beta1 frozen will make it easier to review.

Reply:
This seems like a sub-optimal solution to me. Instead of manually keeping the CRD unchanged, a better approach might have been to bring back the Go types. Now you have a generated CRD that cannot be regenerated anymore, should the need ever arise.

Anyway, just my two cents.

Reply:
I have asked API machinery about this and got confirmation that It is okay to delete the API definitions and only keep the manifests.
If needed, we can always refer back to the go types in earlier branches.